evap/staff/tests/test_forms.py

```diff
@@ -681,3 +681,30 @@ def test_locked_questionnaire(self):
         # Assert form is valid and locked questionnaire is removed
         form.save()
         self.assertEqual({questionnaire}, set(evaluation.general_contribution.questionnaires.all()))
+
+    def test_unused_questionnaire_visibility(self):
+        semester = baker.make(Semester)
+        evaluation = baker.make(Evaluation, course=baker.make(Course, semester=semester))
+        questionnaire = baker.make(Questionnaire, visibility=Questionnaire.Visibility.EDITORS, type=Questionnaire.Type.TOP)
+        questionnaire2 = baker.make(Questionnaire, visibility=Questionnaire.Visibility.EDITORS, type=Questionnaire.Type.TOP)
+        evaluation.general_contribution.questionnaires.add(questionnaire)
+        evaluation.general_contribution.questionnaires.add(questionnaire2)
+
+        form_data = get_form_data_from_instance(EvaluationForm, evaluation)
+
+        self.assertIn(
+            questionnaire.pk,
+            form_data["general_questionnaires"]
+        )
+
+        form_data["general_questionnaires"] = [questionnaire2.pk]
+        form = EvaluationForm(form_data, instance=evaluation, semester=semester)
+        form.save()
+
+        form_data = get_form_data_from_instance(EvaluationForm, evaluation)
+
+        # Now, that no evaluation is using the first questionnaire, it should not be visible anymore
```

Comment:
well. the questionnaire needs to be set to hidden if you want to test the case described in the issue :)
in form_data you will find only selected questionnaires, so you should check for the options or just the existence of the questionnaire's name on the webpage.